### PR TITLE
DDF-3855: Add CSRF Protections to Jolokia and Intrigue Endpoints

### PR DESCRIPTION
**Pull request by garrettfreibott**
#### What does this PR do?
Adds the header "X-Requested-With" as a required header on the backend so that all incoming requests to anything under `/admin/jolokia` or `/search/catalog/internal` must contain this header or a 403 will be returned. Value of this header can be anything, only it's presence is checked. The incoming request must also have either an origin or referer header that matches the target origin or a 403 will be returned. Also adds this header to any necessary UI packages.

#### Who is reviewing it? 
@djblue 
@rzwiefel 
@tbatie 

#### Select relevant component teams: 
@codice/security 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@stustison

#### How should this be tested?
Make sure the admin UI and Intrigue still work. Watch network requests and make sure no 403s are returned. Try accessing jolokia & intrigue internal URLs directly and ensure that they cannot be accessed - a 403 should be returned. See below for a list of packages and contexts that make jolokia and intrigue contexts that should be checked for compatibility.

#### Any background context you want to provide?
Currently holding out on merging this PR while downstream Codice project frontends are checked and modified to ensure compatibility with this new requirement.

The following UI packages under `/ui/packages` make jolokia or intrigue calls and were checked for functionality to ensure that the header was being attached to any outgoing jolokia requests:
`jQuery-updated` ui - /admin
`ok` spatial-geocoding-admin-module - /admin/spatial-config
`ok` resourcemanagement-usage-ui - /admin/data-usage
`ok` resourcemanagement-querymonitor-ui - /admin/queries
`ok` registry-admin-remote-ui - /admin/registry/remote
`ok` registry-admin-local-ui - /admin/registry/local
`ok` catalog-admin-module-sources - /admin/sources
`fetch-updated` catalog-admin-module-maplayers - /admin/map-layers
`fetch-updated` catalog-admin-module-layout - /admin/default-layout
`fetch-updated` broker-undelivered-messages-ui - /admin/undeliveredMessages
`ok` admin-security-certificate-ui - /admin/certificate
`used in test-only` admin-metrics-ui - /admin/metrics
`fetch-updated` admin-core-logviewer - /admin/logviewer

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3855

#### Checklist:
- [x] Documentation Updated
- [X] Update / Add Unit Tests
- [x] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

File: platform/security/filter/security-filter-csrf/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ **/ -->
+<blueprint xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xmlns:cm="http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.1.0"
+           xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0"
+           xsi:schemaLocation="
+  http://www.osgi.org/xmlns/blueprint/v1.0.0 http://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd">
+
+<bean id="filter" class="org.codice.ddf.security.filter.csrf.CsrfFilter">
+    <cm:managed-properties
```
Comment (tbatie) [inline platform/security/filter/security-filter-csrf/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: ❓ do we need the managed-properties here?

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Comment by blen-desta** [comment]
Hero'd successfully 💯

**Comment by clockard** [comment]
build now

**Comment by garrettfreibott** [comment]
buil now

**Comment by garrettfreibott** [comment]
build now

**Comment by clockard** [comment]
Bypassing sonarqube findings as they are not in the change set and are in test classes.

**Pull request merged**